core/io/resource_loader.cpp

```diff
@@ -242,16 +242,18 @@ ResourceLoader::LoadToken::~LoadToken() {
 }
 
 Ref<Resource> ResourceLoader::_load(const String &p_path, const String &p_original_path, const String &p_type_hint, ResourceFormatLoader::CacheMode p_cache_mode, Error *r_error, bool p_use_sub_threads, float *r_progress) {
+	String original_path = p_original_path.is_empty() ? p_path : p_original_path;
 	load_nesting++;
 	if (load_paths_stack->size()) {
 		thread_load_mutex.lock();
-		HashMap<String, ThreadLoadTask>::Iterator E = thread_load_tasks.find(load_paths_stack->get(load_paths_stack->size() - 1));
-		if (E) {
-			E->value.sub_tasks.insert(p_original_path);
+		String parent_task_path = load_paths_stack->get(load_paths_stack->size() - 1);
+		HashMap<String, ThreadLoadTask>::Iterator E = thread_load_tasks.find(parent_task_path);
+		if (E && original_path != parent_task_path) {
```

Comment:
Regarding this comparison, I think it helps because it avoids registering both a resource and its imported counterpart as subtasks, when the former is known to be almost an empty shell. Do you think that's the case? I'm asking because it would be a good idea to add a comment explaining that or, much preferred, make the code explicitly check for that somehow (notwithstanding adding a comment anyway).

Reply:
Yes this is the idea. I can add a comment for that. For the part of checking for that directly I'm not sure how to do that.